x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderTests.java

```diff
@@ -100,12 +99,12 @@ public void testIndexPrefer() {
             d = decider.canAllocate(shard, node, allocation);
             assertThat(node.toString(), d.type(), equalTo(Decision.Type.NO));
             assertThat(node.toString(), d.getExplanation(),
-                containsString("index has a preference for tiers [data_warm,data_cold], " +
+                containsString("index has a preference for tiers [data_warm, data_cold], " +
```

Comment:
Seems like a nit, but I think we don't accept extra whitespace in this setting (we just use `String#split` to parse it) so I don't think we should include it when reporting it back to users either. If we do, someone will copy-paste it as-is and get a message saying `invalid tier names found in [data_hot, data_warm] allowed values are [data_hot, data_warm, ...` which is going to be pretty confusing.

(we should probably fix that error too, I guess almost nobody actually sets these things by hand or else we'd have had a report about it by now)

Reply:
We wouldn't get here with an invalid tier name because of the setting validation that would catch it early. The `validate` logic that you mention in `org.elasticsearch.cluster.routing.allocation.DataTier.DataTierSettingValidator#validate(java.lang.String)` will throw an exception that has the original input string in the "invalid tier names found ". That's why I figured this was ok?

Reply:
Heh this is exactly the confusion I mean: we're saying that this index has a tier preference of `data_warm, data_cold` (with a space) but that's not a valid value for the tier preference setting since we just split at commas, and ` data_cold` (with a leading space) isn't the name of a tier.

Reply:
Bah Github does overly clever stuff with whitespace so my previous comment makes no sense. Reformatting:

    Heh this is exactly the confusion I mean: we're saying that this index has a
    tier preference of `data_warm, data_cold` (with a space) but that's not a valid
    value for the tier preference setting since we just split at commas, and
    ` data_cold` (with a leading space) isn't the name of a tier.

Reply:
Fair point :) I made it use the old style formatting by concatenating strings again (could have used the setting value as well, but I figured this was less-error-prone/more-consistent on the off-chance that we made the setting buggy at some point) and added some early breakouts for the no-debug case so we don't get slower from that.